plugins/module_utils/kibana.py

```diff
@@ -70,4 +71,210 @@ def get_alert_connector_by_name(self, connector_name):
+    pkg_policy_object = pkg_policy_action.get_pkg_policy(agent_policy_id)
+    if not pkg_policy_object:
+      body = {
+        "name": self.pkg_policy_name,
+        "namespace": "default",
+        "description": self.pkg_policy_desc,
+        "enabled": True,
+        "policy_id": agent_policy_id,
+        "output_id": "",
+        "inputs": [],
+        'package': {
+          'name': integration_object['name'],
+          'title': integration_object['title'],
+          'version': integration_object['version']
+        }
+      }
+      body_JSON = dumps(body)
+      endpoint = 'fleet/package_policies'
+      if not self.module.check_mode:
+        pkg_policy_object = self.send_api_request(endpoint, 'POST', data=body_JSON)
+      else:
+        pkg_policy_object = "Cannot proceed with check_mode set to " + self.module.check_mode
+      return pkg_policy_object
+    else:
+      return pkg_policy_object
+    
+# Elastic Agent Policy functions
+
+  def get_all_agent_policys(self):
+    endpoint  = 'fleet/agent_policies'
+    agent_policy_objects = self.send_api_request(endpoint, 'GET')
+    return(agent_policy_objects)
+
+  def create_agent_policy(self, check_mode=False):
+    agent_policy_action = Kibana(self.module)
+    self.agent_policy_id = self.module.params.get('agent_policy_id')
+    self.agent_policy_name = self.module.params.get('agent_policy_name')
+    self.agent_policy_desc = self.module.params.get('agent_policy_desc')
+    if self.agent_policy_id:
+      agent_policy_object = agent_policy_action.get_agent_policy_byid()
+    else:
+      agent_policy_object = agent_policy_action.get_agent_policy_byname()
+      
+    if not agent_policy_object:
+      body = {
+          "name": self.agent_policy_name,
+          "namespace": "default",
+          "description": self.agent_policy_desc,
+          "monitoring_enabled": []
+      }
+      body_JSON = dumps(body)
+      
+      if not self.module.check_mode:
+        endpoint  = 'fleet/agent_policies'
+        agent_policy_object = self.send_api_request(endpoint, 'POST', data=body_JSON)
+        agent_policy_object = agent_policy_object['item']
+      else:
+        agent_policy_object = "Cannot proceed with check_mode set to " + self.module.check_mode
+    else:
+      return
+    return(agent_policy_object)
+
+  def get_agent_policy_byname(self):
+    agent_policy_action = Kibana(self.module)
+    agent_policy_object = ""
+    self.agent_policy_name=self.module.params.get('agent_policy_name')
+    agent_policy_objects = agent_policy_action.get_all_agent_policys()
+    for agent_policy in agent_policy_objects['items']:
+        if agent_policy['name'] == self.agent_policy_name:
+            agent_policy_object = agent_policy
+            continue
+    if agent_policy_object:
+      return(agent_policy_object)
+    else:
+      return
+  
+  def get_agent_policy_byid(self):
+    self.agent_policy_id = self.module.params.get('agent_policy_id')
```

Comment:
Module parameters should be gathered in the module or in `init` if in module_utils as mentioned above.